src/main/java/pro/freeserver/plugin/infrastructure/resources/EconomyRepository.kt

```
@@ -56,11 +67,60 @@ class EconomyRepository(private val client: SupabaseClient) {
+     */
+    suspend fun hasAccount(uuid: UUID): Boolean {
+        try {
+            val freeserverUser = client.postgrest.from("freeserver_user")
+                .select(columns = columns) {
+                    filter {
+                        eq("mcuuid", uuid)
+                    }
+                }
+                .decodeSingle<FreeserverUser>()
+            return freeserverUser.walletBalances != null
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)